Documentation/quickstart/debugging.rst

```diff
@@ -129,7 +129,7 @@ as there are some options hardcoded into
 opencd. By default, it assumes:
 
   * ``CONFIG_DISABLE_MQUEUE=y``
-  * ``CONFIG_PAGING=n``
+  * ``CONFIG_LEGACY_PAGING=n``
```

Comment:
Hi, @xiaoxiang781216 , currently only the NXP LPC31xx boards fully implement the "legacy" on-demand paging to enable larger programs to be loaded (this includes "defconfig"s and board level, like `boards/arm/lpc31xx/ea3152/src/lpc31_boot.c`). However, other SoCs also provide the interfaces for it (like `arch/arm/src/am335x/am335x_boot.c`, although no board-level interfaces are implemented. Greg said it was an experimental feature.

I considered completely removing the legacy support, but I felt it would be reasonable to make it legacy and perhaps wait for a release cycle to make sure no one is using these devices or out-of-tree boards. What do you think? Should we remove it now?

Reply:
@xiaoxiang781216 @tmedicci I think we shouldn't destroy the legacy paging until we confirm everything is working as previously. I think this new implementation is superior, but maybe there are case where the older is better or necessary. AFAIK the new implementation doesn't work on FLAT mode (at least the documentation says it), to there still place to older implementation. If this is the case, maybe instead legacy, it could be renamed to CONFIG_FLATMODE_PAGING or something like this!

Reply:
OK